go-controller/observability-lib/sampledecoder/sample_decoder.go

```diff
@@ -0,0 +1,290 @@
+package sampledecoder
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/ovn-org/libovsdb/client"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/observability-lib/ovsdb"
+	libovsdbops "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdb/ops"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/observability"
+)
+
+type SampleDecoder struct {
+	nbClient          client.Client
+	ovsdbClient       client.Client
+	cleanupCollectors []int
+}
+
+type nbConfig struct {
```

Comment:
We use this for both NB and OVS DB, I think a better name is: `type dbConfig struct`